File: rudi-microservice/rudi-microservice-strukture/rudi-microservice-strukture-core/src/main/java/org/rudi/microservice/strukture/core/bean/criteria/OrganizationSearchCriteria.java
package org.rudi.microservice.strukture.core.bean.criteria;

import java.util.List;
import java.util.UUID;

import org.rudi.bpmn.core.bean.Status;
import org.rudi.common.core.bean.criteria.SearchCriteria;
import org.rudi.microservice.strukture.core.bean.OrganizationStatus;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import lombok.experimental.SuperBuilder;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@SuperBuilder
@ToString
public class OrganizationSearchCriteria implements SearchCriteria {
	private List<UUID> uuids;
	private String name;
	private Boolean active;
	private UUID userUuid;
	private List<OrganizationStatus> organizationStatus;
	private Status status;
	private Boolean loadAllInformations;
	private List<UUID> excludeOrganizationUuids ;

	private Integer offset;
	private Integer limit;
	private String order;

	List<OrganizationStatus> adminMemberAllowedStatus;
}

File: rudi-application/rudi-application-front-office/angular-project/src/app/core/bean/strukture/organization-search-criteria.ts
import {SearchCriteria} from '@core/bean/search-criteria';
import {Status} from 'micro_service_modules/api-bpmn';
import {OrganizationStatus} from 'micro_service_modules/strukture/strukture-model';

export class OrganizationSearchCriteria extends SearchCriteria {
    uuid?: string;
    name?: string;
    active?: boolean;
    userUuid?: string;
    organizationStatus?: OrganizationStatus;
    status?: Status;
}
